### PR TITLE
format single-digit cents properly

### LINKED ISSUE #13: Auto-format cents on blur
cf https://github.com/dobtco/screendoor-v2/issues/1183

When the cents field is blurred, if there is exactly one digit, prefix it with a `0`.

E.g. "3" -> "03"

### PR DESCRIPTION
**Pull request by aviv**
@ajb will close #13 

Do we also want to format this on the server? I.e. we should probably also convert "3" to "03" instead of "30" on the server, too, otherwise "3" would still get interpreted as "30" if the user has javascript disabled.

### CODE REVIEW

File: src/templates/fields/price.eco [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <div class='fr_item_auto'>
       <input type="text"
              data-rv-input='model.value.cents'
+             data-js-price-cents
```
Comment (ajb) [inline src/templates/fields/price.eco]: Can we just leave out this extra attribute, and select  this input using `[data-rv-input=model.value.cents]`?

File: src/views.coffee [hunk 2]
```diff
@@ -112,6 +112,17 @@ FormRenderer.Views.NonInputResponseField = FormRenderer.Views.ResponseField.exte
     @$el.html JST['partials/non_input_response_field'](@)
     @
 
+FormRenderer.Views.ResponseFieldPrice = FormRenderer.Views.ResponseField.extend
+  field_type: 'price'
+  events:
+    'blur [data-js-price-cents]': 'formatCents'
+
+  formatCents: ->
+    @$input = @$el.find('input[data-js-price-cents]')
```
Comment (ajb) [inline src/views.coffee]: If you pass the `event` to this function, you should be able to get the DOM node via `event.target`.

**Comment by ajb** [comment]
You also need to run `grunt && grunt dist`. (You must know this... otherwise you wouldn't see your changes anywhere?) 

And see if you can add a test, too.

**Comment by ajb** [comment]
Not sure what you mean by formatting on the server, but FWIW, if a user has JavaScript disabled, they won't be able use any of our apps.

**Comment by aviv** [comment]
When I run `grunt && grunt dist`, I get this error when trying to load the page:


I have to discard changes to `vender.js` to get it to work. Maybe my dev environment for `formrenderer-base` isn't quite set up properly? Spent a while trying to get it to work. I pushed what I get after running `grunt && grunt dist`.

**Comment by aviv** [comment]
> Not sure what you mean by formatting on the server

Just that if the user somehow manages to get "3" in the cents field without this JS formatter converting it to "03", we'll interpret it incorrectly as "30 cents" (cf. the original issue: dobtco/screendoor-v2#1183). But if we're fine relying on the client to format this properly, no worries.

**Comment by ajb** [comment]
1. I think you need to `bower install` before running any grunt tasks.
2. Yes, you're right about the way that the server interprets the value. I'll re-open https://github.com/dobtco/screendoor-v2/issues/1183.

**Comment by aviv** [comment]
That was it, thanks! Was doing `npm install` but didn't consider `bower` :| -- guess I'm getting to comfy in the rails environment, and need to read up a bit on backbone etc.

**Comment by aviv** [comment]
looks like jquery is getting bumped, which is why the diff is weird (`vendor.js`). do i need to separate this out, or is it good to merge?

**Comment by ajb** [comment]
Nope, this is fine. I'll review later today.

File: src/views.coffee [hunk 3]
```diff
@@ -112,6 +112,17 @@ FormRenderer.Views.NonInputResponseField = FormRenderer.Views.ResponseField.exte
     @$el.html JST['partials/non_input_response_field'](@)
     @
 
+FormRenderer.Views.ResponseFieldPrice = FormRenderer.Views.ResponseField.extend
+  field_type: 'price'
+  events:
+    'blur [data-rv-input="model.value.cents"]': 'formatCents'
+
+  formatCents: (ev) ->
+    @$input = $(ev.target)
```
Comment (ajb) [inline src/views.coffee]: 1) the best-practice is to name the event parameter `e`, not `ev`.
2) you're setting `@$input` as an instance variable, when it should just be a local

**Pull request merged**